### DISCUSSION TITLE
Reinstalled app / new phone, matching everything again ?

### DISCUSSION THREAD

**Discussion by traktuner** [immich-app/immich#4175] [Q&A]
Hello!
I am on iOS and have a large photo library of about 20k pictures/videos.
I switched phone, have the same photo library, but it goes through every picture/video again and checks if it's already uploaded. Is there a way to tell the app to only back up "new" assets from when the app got reinstalled (or make the matching process way faster)?
It takes about 3 days and nights that the phone goes through all items (and it get's really, really hot during that time)
Thank you!

**Comment by alextran1502** [comment]
Hello. Before we work on a mechanism to address this issue, the only way to overcome this will be to change the database record.

**Reply by traktuner** [reply]
Thanks for your reply! Are there instructions in the docs how I can do that? Just onboarded some users with existing libraries...

**Reply by srwareham** [reply]
Alex, would you be able to elaborate on the process to 'change the database record' ? I expect this will be a pretty common edge case in the holiday season as people get new phones.

**Reply by HybridSarcasm** [reply]
Hi @alextran1502 : [a recent message from you on Discord](https://discord.com/channels/979116623879368755/1177969608230183022/1178167203695169556) suggests this is on the roadmap. Is there a release milestone for this feature yet?

**Reply by alextran1502** [reply]
We don't have ETA at the moment

**Reply by id-ic** [reply]
Hi @alextran1502 I'm assuming this hasn't been addressed yet but I ran into the same issue rebuilding my iPhone.  I see there is a comment from back in Nov of 2023 about it being on the roadmap.  What item is it on the road map?

As of now, i'm guessing the only way around this is to upload everything again and have the duplicate process flag them?  Is there a way to mass delete duplicates?

thanks for the great app!

**Reply by bo0tzz** [reply]
> What item is it on the road map?

Better background backups

> Is there a way to mass delete duplicates?

There is in the deduplication utility, but these assets would be exact duplicates and so wouldn't get saved after upload at all.

**Reply by id-ic** [reply]
Thanks for the response @bo0tzz

**Comment by ben-z** [comment]
I have a similar issue where I'm installing immich onto multiple devices for the first time. They all share the same iCloud Photo library with about 60K photos/videos. All of the devices appear to be uploading all of the photos from scratch. The server deduplicates but uploading is slow. I have also already uploaded the entire library using CLI.

@traktuner how did you get continuous uploading work? I have both foreground and background backup enabled, but the upload don't seem to progress fast enough. It's been about 2 weeks and the fastest device only got through ~10K assets. That device is constantly connected to the charger too. The network latency and bandwidth from the device to the server is pretty good too (~50Mbps, 20ms ping).

**Comment by traktuner** [comment]
@ben-z 
Hey,
I manually imported all pictures/videos via command line in docker, after that, the phone just matches the hashes of the images - it seems that it's uploading something but it doesn't and that goes through many assets in one second. I also had to wait a while since it re-uploaded some big movies but it went quicker than uploading the while library initially.
Also, server is in my local network, so much faster.
Hope that helps!

**Comment by blublub** [comment]
Same problem here now. New phone and Immich wants  re-upload all 3000+ images although only like 10-15 are actually "new".

This means that Immich will re-upload all of them which takes a considerable amount of time and that I will have images on the server re-appear which I have deleted prior - the latte really sucks.

What about a hash check?

So that the Immich app on the phone runs a tasks to create hashes and then uses the hashes to check which images are already present on the server?

**Comment by pug306d** [comment]
Same problem for me now, my photo library has 46000 images and videos which are all stored on iCloud not my new iPhone and Immich is trying to download and upload everything again 😢

The new iPhone was restored from a backup of the previous iPhone - could the states not be stored with the app and carry forward to new devices provided it was restored from an existing iCloud backup?

**Comment by leah-potato** [comment]
Same issue for me. iPhone went for a swim, and I thought to myself no problem - I already had Immich all setup. It’s now redownloading and uploading 20k items 🤦‍♀️

Isn’t this scenario exactly why someone would have Immich in the first place?

**Comment by Stooovie** [comment]
Still no news on this? I had to clean install my iPad I use for syncing Immich with iCloud Photo Library so now I’m looking at resyncing 60k assets again for no apparent reason.

**Reply by YOZZOPANOZZO** [reply]
I'm on the same boat. It seems like it doesn't check the hash before uploading something... I am on Android.

**Reply by TiggsPanther** [reply]
This issue, and the threads I found about it, were what ended me testing Immich.  

My up-to-recent iCloud library is already downloaded and sat on my NAS. Setting it as an external library was easy.  
Having it then try to re download and duplicate upload my library was not something that interested me. And having to go through the same, and more, when I next upgrade my phone was a showstopper.

**Reply by srwareham** [reply]
For what it's worth, I've seen many commits in the immich repo related to syncing lately. If I was a gambling man I'd bet a fix for this is coming in the next release. This falls under [better background backups](https://immich.app/roadmap/) in the roadmap, which is a key milestone towards the stable version. It's been long enough since the last release that the next one could even potentially be the first stable one.

**Reply by cjlapao** [reply]
This just happened to me, same phone, on iOS what changed was moving to iOS 26, still this is something that will make me go back to google photos, they are too precious to be playing with it

**Reply by TiggsPanther** [reply]
Dammit!  

Looks like I'll have to either take the app out of the equation and separately upload all my photos to Immich and iCloud or just drop Immich entirely.

Glad to hear about this _before_ I updated my phone to iOS 26, rather than after.

**Reply by cjlapao** [reply]
I think the worst part is even if you allow it to go through everything and resync because the database will not allow the keys to be inserted, so it is now on this weird loop of attempting to resync everything, finishes it and then restarts the resync. uninstalling it for now until we have a fix
